epidose/device/util.sh

```diff
@@ -186,6 +196,24 @@ get_new_filter()
   fi
 }
 
+# Check and update software if necessary
+# preconditions: WiFi should be turned on
+check_for_updates()
+{
+  log "Checking for updates"
+  remote_updates_available=false
+ 
+  # Set username and email because git asks for it
+  git config --system user.name "$USERNAME"
+  git config --system user.email "$EMAIL"
+  git -C "$LOCAL_REPO" remote update && git -C "$LOCAL_REPO" status -uno | grep -q 'Your branch is behind' && remote_updates_available=true
+  if [ "$remote_updates_available" = true ]; then
+    git -C "$LOCAL_REPO" pull "$REMOTE_REPO" && exec ./"$LOCAL_REPO"/epidose/device/update.sh
```

Comment:
Will the `exec` terminate the `update_filter_d`?  Also the please take care of file naming: `update_filter_d` now does more things, and it is unclear what `update.sh` does.

Reply:
There's a TODO a couple of lines below, right?

Reply:
1. The `exec` is not terminating the `update_filter_d`, but I think the `supervisor` does.
2. I have named the `update.sh` to `update_software.sh`.
3. I guess you mean in the `update_filter` where it says check for software updates. But we acquire the Wifi before the TODO comments. Should I still check for updates there or remote those comments?

Reply:
Remove the outdated comments.

Reply:
Done!